plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/utils/RenderingPreparer.xtend

```diff
@@ -129,6 +130,45 @@ final class RenderingPreparer {
                 prepareRendering(port)
             }
         }
+        
+        // Also calculate the sizes of all proxy-renderings
+        val proxyRendering = element.getProperty(KlighdProperties.PROXY_VIEW_PROXY_RENDERING)
+        if (element.getProperty(KlighdProperties.PROXY_VIEW_RENDER_NODE_AS_PROXY) && proxyRendering !== null) {
+            for (data : proxyRendering) {
+                switch(data) {
+                    KRenderingLibrary: {
+                        // The library needs to generate ids for all later KRenderingRefs to refer to, but no own bounds,
+                        // since these are generic renderings.
+                        for (rendering : data.renderings) {
+                            if (rendering instanceof KRendering) {
+                                KRenderingIdGenerator.generateIdsRecursive(rendering)
+                            }
+                        }
+                    }
```

Comment:
I guess there should never be a rendering library hidden in some proxy. The library is only ever to be defined in the data of the top level graph element.